### PR TITLE
Add `fileType.extensions` and `fileType.mimeTypes` properties

### LINKED ISSUE #91: isSupported() method
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/file-type/issues/91)

It'd be nice to have a function that returns if input extension (string) is supported by the module

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $20.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($20.00)
### Submitted pull Requests
- [#229 added fileType.extensions and fileType.mimeTypes properties](https://issuehunt.io/r/sindresorhus/file-type/pull/229)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/file-type/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by adityapatadia**
closes #91

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#91: isSupported() method](https://issuehunt.io/repos/19105307/issues/91)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

**Comment by sindresorhus** [comment]
> If anyone wants to work on this, see the feedback in https://github.com/sindresorhus/file-type/pull/209

You did not address all the feedback from the previous PR.

**Comment by sindresorhus** [comment]
You need to add the `supported.js` file to `files` in package.json.

**Comment by adityapatadia** [comment]
> > If anyone wants to work on this, see the feedback in #209
> 
> You did not address all the feedback from the previous PR.

Which specific feedback? I did read all comments.

**Comment by adityapatadia** [comment]
> You need to add the `supported.js` file to `files` in package.json.

Done

**Comment by sindresorhus** [comment]
> You also need to add tests and **add it to /index.d.ts@master** - https://github.com/sindresorhus/file-type/pull/209#issuecomment-489311870

**Comment by adityapatadia** [comment]
> You also need to add tests and **add it to /index.d.ts@master** - [#209 (comment)](https://github.com/sindresorhus/file-type/pull/209#issuecomment-489311870)

Done.

**Comment by sindresorhus** [comment]
> add it to /index.d.ts@master - #209 (comment)

`index.d.ts`, not just `index.test-d.ts`.

**Comment by adityapatadia** [comment]
Oops, I don't know much about typescript. I will correct.

**Comment by adityapatadia** [comment]
Added it.

File: index.d.ts [hunk 1]
```diff
@@ -173,6 +173,18 @@ declare const fileType: {
 	The minimum amount of bytes needed to detect a file type. Currently, it's 4100 bytes, but it can change, so don't hard-code it.
 	*/
 	readonly minimumBytes: number;
+	
+	/** 
+	Returns set of extensions supported by this library.
+	*/
+	
+	readonly extensions: object;
```
Comment (sindresorhus) [inline index.d.ts]: This is not correct. It should be of type `FileType`. You also need to create a type for `mimeTypes`.

File: index.d.ts [hunk 2]
```diff
@@ -173,6 +173,18 @@ declare const fileType: {
 	The minimum amount of bytes needed to detect a file type. Currently, it's 4100 bytes, but it can change, so don't hard-code it.
 	*/
 	readonly minimumBytes: number;
+	
+	/** 
+	Returns set of extensions supported by this library.
```
Comment (sindresorhus) [inline index.d.ts]: It doesn't return anything. It's a property. It *is* that value.

File: index.d.ts (hunk 1 above)
Comment (sindresorhus) [reply index.d.ts]: You also need to update https://github.com/sindresorhus/file-type/blob/master/.github/pull_request_template.md

**Comment by adityapatadia** [comment]
Please check now.

**Comment by adityapatadia** [comment]
I am not able to understand typescript error. Can you please check?

**Comment by adityapatadia** [comment]
It's fixed now.

File: index.d.ts [hunk 3]
```diff
@@ -157,28 +258,40 @@ declare const fileType: {
 
 	const url = 'https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif';
 
-	http.get(url, response => {
-		response.on('readable', () => {
+	http.get(url response => {
+		response.on('readable' () => {
```
Comment (bencmbrook) [inline index.d.ts]: Are all these removed commas a bad find-and-replace?

Comment (adityapatadia) [reply index.d.ts]: Yes, reverted it.

**Comment by adityapatadia** [comment]
Done all changes.

**Comment by adityapatadia** [comment]
@sindresorhus please check this as well. If this can be merged.

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/file-type/pull/229/files#diff-b52768974e6bc0faccb7d4b75b162c99R229 should use the new Mime TS type.

**Comment by sindresorhus** [comment]
While I appreciate the PR, this PR was too sloppy. It was a very easy issue and it even has a bounty, which means I expect more. I spent 6 times more time just reviewing this than it would have taken me to just do the change myself. The point of bounties is to save me time. So, all I'm asking is that to put some more effort into future PRs ;)

**Pull request merged**